packages/flutter/lib/src/widgets/text.dart

```diff
@@ -1212,6 +1223,91 @@ class _SelectableTextContainerDelegate extends MultiSelectableSelectionContainer
     return a.right > b.right ? 1 : -1;
   }
 
+  @override
+  SelectedContent? getSelectedContent() {
+    if (currentSelectionStartIndex == -1 || currentSelectionEndIndex == -1) {
+      return null;
+    }
+    final List<SelectedContent> selections = <SelectedContent>[
+      for (final Selectable selectable in selectables)
+        if (selectable.getSelectedContent() case final SelectedContent data) data,
+    ];
+    if (selections.isEmpty) {
+      return null;
+    }
+    // A [_SelectableFragment] does not typically provide a [SelectedContentController],
+    // so this pass should collect all [PlaceholderSpan] child controllers.
+    final List<SelectedContentController<Object>> childControllers = <SelectedContentController<Object>>[
+      for (final SelectedContent selectedContent in selections)
+        if (selectedContent.controllers case final List<SelectedContentController<Object>> data) ...data,
+    ];
+    // Question: Should we add the root text controller, if only the children
+    // are selected? For example, say we have the following:
+    // Text.rich(
+    //   TextSpan(
+    //     text: 'This is some bulleted list:\n',
+    //     children: <InlineSpan>[
+    //       WidgetSpan(
+    //         child: Column(
+    //           children: <Widget>[
+    //             for (int i = 1; i <= 7; i += 1)
+    //               Padding(
+    //                 padding: const EdgeInsets.only(left: 20.0),
+    //                 child: Text('• Bullet $i'),
+    //               )
+    //           ],
+    //         ),
+    //       ),
+    //     ],
+    //   ),
+    // ),
+    // and we select only text inside the widgetspan, should we return the
+    // root text controller in this case, as well as the children controllers
+    // that are selected? Or should we only return the selected children controllers.
+    // The former might be more confusing to the user?
```

Comment:
What if a user wants to change something in the part of the text that is not selected? If we want to support that use case then I think we'd have to return the root controller.